### PR TITLE
feat: add fs_version and compressor output of nydus image check

### PR DESCRIPTION
**Pull request by PerseidMeteor**
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
1. Add fs_version value, output v5 or v6.
2. Add compressor algorithm value, like ztsd.
3. Add rafs_version and compressor JSON output of nydus image check, so that more info can be get if it is necessary.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/bin/nydus-image/main.rs [hunk 1]
```diff
@@ -114,6 +114,8 @@ impl OutputSerializer {
         build_info: &BuildTimeInfo,
         blob_ids: Vec<String>,
         bootstrap: &Path,
+        compressor: String,
+        rafs_version: u32,
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:118]: Should name it `fs_version`.

File: src/bin/nydus-image/main.rs [hunk 2]
```diff
@@ -126,16 +128,14 @@ impl OutputSerializer {
...
-                trace,
-            };
-
-            serde_json::to_writer(w, &output).context("failed to write result to output file")?;
+            let mut out = serde_json::Map::new();
+            out.insert("version".to_string(), serde_json::Value::from(format!("{}-{}", build_info.package_ver, build_info.git_commit)));
+            out.insert("bootstrap".to_string(), serde_json::Value::from(bootstrap.display().to_string()));
+            out.insert("blobs".to_string(), serde_json::Value::from(blob_ids));
+            out.insert("trace".to_string(), serde_json::Value::from(root_tracer!().dump_summary_map().unwrap_or_default()));
+            out.insert("compressor".to_string(), serde_json::Value::from(compressor));
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:136]: Let's we define the fields in `OutputSerializer`.

File: src/bin/nydus-image/main.rs (hunk 1 above)
Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: renamed

File: src/bin/nydus-image/main.rs (hunk 2 above)
Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: how about `fs_version` and `compressor`

Comment (imeoer) [reply src/bin/nydus-image/main.rs]: > how about `fs_version` and `compressor`

Should be ok.

File: src/bin/nydus-image/main.rs [hunk 3]
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
     blobs: Vec<String>,
     /// Performance trace info for current build.
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v5 or v6).
+    fs_version: u32,
+    /// compress algorithm.
```
Comment (Desiki-high) [inline src/bin/nydus-image/main.rs:82]: Capitalization: `Compress algorithm.`

Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: resolved

File: src/bin/nydus-image/main.rs [hunk 4]
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
     blobs: Vec<String>,
     /// Performance trace info for current build.
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v4, v5 or v6).
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:80]: `Rafs version(v4, v5 or v6).` -> `Rafs version(5 or 6).`

File: src/bin/nydus-image/main.rs [hunk 5]
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
...
+    fs_version: u32,
+    /// Compress algorithm.
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:82]: `Chunk compression algorithm`

File: src/bin/nydus-image/main.rs (hunk 4 above)
Comment (imeoer) [reply src/bin/nydus-image/main.rs:80]: In fact have no v4 for upstream codes. :)

File: src/bin/nydus-image/main.rs [hunk 6]
```diff
@@ -95,11 +104,19 @@ impl OutputSerializer {
                 .with_context(|| format!("can not open output file {}", f.display()))?;
             let trace = root_tracer!().dump_summary_map().unwrap_or_default();
             let version = format!("{}-{}", build_info.package_ver, build_info.git_commit);
+            let out_fs_version: u32 = match fs_version {
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:107]: Let's define `fs_version: RafsVersion` and use `RafsVersion::to_string()` here.

Comment (imeoer) [reply src/bin/nydus-image/main.rs:107]: Let's define `compressor: compress::Algorithm` and use `compress::Algorithm::to_string()` here.

Comment (imeoer) [reply src/bin/nydus-image/main.rs:107]: Or implement JSON Serialize for `RafsVersion` and `compress::Algorithm`.

Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: we can use `to_string` method

File: src/bin/nydus-image/main.rs (hunk 4 above)
Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: fine

File: src/bin/nydus-image/main.rs [hunk 7]
```diff
@@ -74,13 +74,19 @@ pub struct OutputSerializer {
     blobs: Vec<String>,
     /// Performance trace info for current build.
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v5 or v6).
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:77]: Should be `5 or 6`

**Review by imeoer — COMMENTED** [review]
other LGTM, thanks!

File: src/bin/nydus-image/main.rs (hunk 7 above)
Comment (PerseidMeteor) [reply src/bin/nydus-image/main.rs]: resolved

**Review by adam3q — COMMENTED** [review]
Thanks! Could you update the `nydus-image check` usage in `docs/nydus-image.md` ?

**Review by imeoer — APPROVED** [review]
Thanks!

**Comment by PerseidMeteor** [comment]
> Thanks! Could you update the `nydus-image check` usage in `docs/nydus-image.md` ?

`docs/nydus-image.md`  will be updated later.

**Pull request merged**